### PR TITLE
`rust_sitrep()`: explain what toolchains are required, guide through installation

### PR DESCRIPTION
**Pull request by Ilia-Kosenkov**
- Add color-coding to `rust_sitrep()`, highlighting required (and installed) toolchains/targets with green.
- Check if required targets/toolchains are missing, notify the user if they are. Provide guidelines on how to install missing components.

Here are examples of prompts:

### CODE REVIEW

File: README.Rmd [hunk 1]
```diff
@@ -54,6 +54,18 @@ To execute Rust code, you will also need to set up a working Rust toolchain. See
 
 ## Usage
 
+The status of Rust toolchain and available targets can be checked with `rust_sitrep()`:
+
+```{r eval=FALSE, echo=TRUE}
```
Comment (JosiahParry) [inline README.Rmd:59]: I think this part can be expanded by 1 or 2 sentences. Can we suggest this as the first step for new users to run to ensure that they have a compatible version of Rust installed. If not, the function will guide them on how to install the appropriate tool chain?

File: README.Rmd [hunk 2]
```diff
@@ -54,6 +54,18 @@ To execute Rust code, you will also need to set up a working Rust toolchain. See
...
+rust_sitrep()
+# Rust infrastructure sitrep:
+# ✔ "rustup": 1.26.0 (5af9b9484 2023-04-05)
+# ✔ "cargo": 1.72.0 (103a7ff2e 2023-08-15)
+# ℹ host: x86_64-pc-windows-msvc
+# ℹ toolchain: stable-x86_64-pc-windows-msvc (default)
+# ℹ target: x86_64-pc-windows-gnu
+```
```
Comment (JosiahParry) [inline README.Rmd:67]: perhaps, if you know how to repro, can we provide what the output of a _failed_ sitrep would look like so users know how to continue from there?

**Review by JosiahParry — COMMENTED** [review]
Genuinely is a very clean PR. The only request I have is slight improvements to the README.

~ pretends to select "Request changes" ~

**Comment by JosiahParry** [comment]
Changes look great!

**Comment by Ilia-Kosenkov** [comment]
Can you approve this PR or you don't have the permissions?

**Comment by JosiahParry** [comment]
No permissions but consider this my approval 👍🏼 
😉

**Review by CGMossa — APPROVED** [review]
LGT @JosiahParry , so LGTM

**Comment by CGMossa** [comment]
You can leave a "community member" review, and honestly those do help.

I left one for another project today as a community member.. 🤷‍♂️

**Pull request merged**